High_Level_Introduction_to_ASDF.ipynb

```diff
@@ -0,0 +1,214 @@
+   "source": [
+    "## High Level ASDF File Structure"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "An ASDF file consists of a single YAML header, followed by zero or more binary blocks.\n",
+    "\n",
+    "YAML (Yet Another Markup Language) is a standard text format that is easier to read than XML and more concise and readable than JSON. It is generally intended that the YAML header will hold all the metadata for the data contained within the file. It also is used to indicate the organization of all the data and how it is associated with relevant metadata."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Added Features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- No limit on the length of attribute names (analogous to FITS keyword names).\n",
+    "- No limit on attribute values, which may theselves be complex values. This allows arbitrarily nested structures.\n",
+    "- Ability to \"tag\" values as having a special type that software may know how to specially handle.\n",
+    "- Ability to have different attributes reference the same data or object without making copies, so that data or object can be shared by many things.\n",
+    "- Full support for all standard numerical binary values.\n",
+    "- Supports mode of putting some or all data \"in-line\", i.e., as text values.\n",
+    "- Supports very flexible WCS model.\n",
+    "- Mechanism for extensions to the standard format, both in the standard itself, and for local purposes.\n",
+    "- Versioned globally, and for specific extensions.\n",
+    "- Support for schema files that can be used to validate the proper structure of the file, both for reading and writing.\n",
+    "- Support for streamed data.\n",
+    "- Language agnostic, but currently only fully supported in Python\n",
+    "\n",
+    "Many of these items value may not yet be obvious to the reader, but tutorials will illustrate  the value with examples."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## FITS Features not yet Supported"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Variable length records in tables (using heap storage)\n",
+    "- Random Groups"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites to Tutorials\n",
+    "\n",
+    "### Installations\n",
+    "\n",
+    "Of course Python 3.6 or later must be installed, along with ipython, jupyter, numpy, astropy, gwcs, and asdf. We generally recommend installing Python using miniconda since conda allows easy cloning of different Python environments ([miniconda download](https://docs.conda.io/en/latest/miniconda.html) and using pip install for the rest after activating the miniconda environment. For example, at the shell level:\n",
+    "\n",
+    "`\n",
+    "pip install ipython\n",
```

Comment:
If that is the normal solution, yes, I'll do that.

Reply:
Though I was chastised on slack for recommending pip. For most things  it seems to work fine. I probably wouldn't use in an operational case. Should I also suggest setting up a special environment just for the tutorial so that pip doesn't mess up some existing installation? Yes, probably.